README.md

```diff
@@ -78,11 +65,31 @@ If set, this value is prepended at the start of the record name.
 
 ### Record name separator
 If set, this value is inserted between each component of the record name (record number, DAG name,
-user supplied). If only one component is used, this value is ignored.
+user supplied etc). If only one component is used, this value is ignored.
+
+Note that a constant value (if used) is not separated from its adjacent components by the separator,
+so it will have to be included explicitly in the constant value if required.
 
 ### Record name suffix
 If set, this value is appended at the end of the record name.
 
+### Record numbering
+This defines how the records in this arm are numbered.
+
+* If the **arm** option is selected, then per-arm numbering is used. This means that a separate
+  record counter will be used for the records in this arm, so they will be numbered from the arm's
+  first record number and will not be affected by records in other arms (so long as there is not a
+  naming conflict). If this option is not selected, then each new record will use the next available
+  number project-wide.
+* If the other options are selected, a separate record counter will be used for records which share
+  that component of the record name. This will allow numbering based on elements of the record name
+  (e.g. per-DAG, per-timestamp, per-user), either individually or in any combination. The numbering
+  can be shared between arms if per-arm numbering is disabled and the other arms share the
+  configuration.
+* Take care to avoid naming conflicts if per-arm numbering is used. Setting a distinct prefix or
+  suffix for the record name on each arm is recommended.
+* Record numbering cannot be set if a record number is not used.
+
```

Comment:
I always wonder if the zero pad number is the number  of zeros to put, or total number length.  So if zero pad is 3 would we have a number 0001 and 0123 or 001 and 123.  Perhaps an example in the readme?

Reply:
Zero pad 3 digits = number will be 3 digits after zero padding (e.g. 1 => 001).